core/gui/CMakeLists.txt

```diff
@@ -43,6 +43,14 @@ ROOT_OBJECT_LIBRARY(GuiCore
   src/TToggleGroup.cxx
 )
 
-# target_include_directories(GuiCore PRIVATE res ../foundation/res)
+target_include_directories(GuiCore PRIVATE 
+   ${CMAKE_SOURCE_DIR}/core/base/inc
+   ${CMAKE_SOURCE_DIR}/core/clib/inc
+   ${CMAKE_SOURCE_DIR}/core/meta/inc
+   ${CMAKE_SOURCE_DIR}/core/cont/inc
```

Comment:
This must all be picked up by linking against `-lCore`. No Core includes must be enumerated here or in any other module.

Reply:
But `GuiCore` is not linked against `Core` - it is part of `Core` and defined **before**. Therefore one cannot use target properties, which is not yet there. Or again - one should try `cmake generator expressions`